.github/workflows/docker-publish.yml

```diff
@@ -28,71 +28,67 @@ on:
     tags: [ 'v*.*.*' ]
   pull_request:
     branches: [ main ]
-
 env:
-  # Use docker.io for Docker Hub if empty
   REGISTRY: ghcr.io
-  # github.repository as <account>/<repo>
-  REGISTRY_SERVER: ${{github.repository_owner}}/registry-server
-  # github.repository as <account>/<repo>
-  REGISTRY_ENVOY: ${{github.repository_owner}}/registry-envoy
-
-
+  USERNAME: ${{ github.actor }}
+  PASSWORD: ${{ secrets.GITHUB_TOKEN }}
+  DOCKER_REPOSITORY_OWNER: ${{github.repository_owner}}
 jobs:
-  build:
-
+  build-registry-server:
     runs-on: ubuntu-latest
-    permissions:
-      contents: read
-      packages: write
-
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v2
-
-      # Login against a Docker registry except on PR
-      # https://github.com/docker/login-action
-      - name: Log into registry ${{ env.REGISTRY }}
-        if: github.event_name != 'pull_request'
-        uses: docker/login-action@28218f9b04b4f3f62068d7b6ce6ca5b26e35336c
-        with:
-          registry: ${{ env.REGISTRY }}
-          username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - uses: ./.github/workflows/composite/build-push
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ env.USERNAME }}
+        password: ${{ env.PASSWORD }}
+        image: ${{ env.DOCKER_REPOSITORY_OWNER }}/registry-server
+        context: .
+        file: containers/registry-server/Dockerfile
 
-      # Extract metadata (tags, labels) for Docker
-      # https://github.com/docker/metadata-action
-      - name: Extract Docker metadata
-        id: meta
-        uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
-        with:
-          images: ${{ env.REGISTRY }}/${{ env.REGISTRY_SERVER }}
+  build-registry-tool:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+    - uses: ./.github/workflows/composite/build-push
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ env.USERNAME }}
+        password: ${{ env.PASSWORD }}
+        image: ${{ env.DOCKER_REPOSITORY_OWNER }}/registry-tool
+        context: .
+        file: containers/registry-tool/Dockerfile
```

Comment:
Shouldn't this be "registry-tools"? (it's plural because it contains the `registry` tool and the automatically-generated `apg` tool)

Reply:
I changed it back , realized it has multiple tools